src/bsp/src/stm32/src/Decawave.cpp

```diff
@@ -272,4 +289,49 @@ void Decawave::retrieveRxFrame(UWBRxFrame* frame) {
     } else if ((m_callbackData.status & SYS_STATUS_ALL_RX_ERR) != 0U) {
         frame->m_status = UWBRxStatus::ERROR;
     }
-}
\ No newline at end of file
+}
+
+void Decawave::setTxAntennaDLY(uint16 delay) {
+    deca_selectDevice(m_spiDevice);
+    dwt_settxantennadelay(delay);
+}
+
+void Decawave::setRxAntennaDLY(uint16 delay) {
+    deca_selectDevice(m_spiDevice);
+    dwt_setrxantennadelay(delay);
+}
+
+void Decawave::getTxTimestamp(uint64_t* txTimestamp) {
+    deca_selectDevice(m_spiDevice);
+    dwt_readtxtimestamp((uint8_t*)txTimestamp);
+}
+
+void Decawave::getRxTimestamp(uint64_t* rxTimestamp) {
+    deca_selectDevice(m_spiDevice);
+    dwt_readrxtimestamp((uint8_t*)rxTimestamp);
+}
+void Decawave::getSysTime(uint64_t* sysTime) {
+    deca_selectDevice(m_spiDevice);
+    dwt_readrxtimestamp((uint8_t*)sysTime);
+}
+
+void Decawave::finalMsgAddTs(uint8* tsField, uint64_t ts) {
```

Comment:
This isn't only for the final message, it is a utils method to convert a TS to it's binary value and it should be named as such (and as I said in the .h, not sure this belongs here)